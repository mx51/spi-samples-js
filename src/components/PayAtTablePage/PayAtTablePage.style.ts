import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const usePayAtTableStyles = makeStyles((theme: Theme) =>
  createStyles({
    container: {
      paddingTop: theme.spacing(7),
      paddingBottom: theme.spacing(7),
    },
    addButton: {
      position: 'absolute',
      right: theme.spacing(7),
      bottom: theme.spacing(7),
    },
    cardContainer: {
      maxWidth: '300px',
    },

    iconContainer: {
      display: 'flex',
    },
    successIcon: {
      color: theme.palette.success.main,
      margin: theme.spacing(1),
      marginLeft: 'auto',
      marginRight: 'auto',
      width: '20px',
      height: '20px',
    },
    failedIcon: {
      color: theme.palette.error.main,
      margin: theme.spacing(1),
      marginLeft: 'auto',
      marginRight: 'auto',
      width: '20px',
      height: '20px',
    },
    unknownIcon: {
      color: theme.palette.warning.main,
      margin: theme.spacing(1),
      marginLeft: 'auto',
      marginRight: 'auto',
      width: '20px',
      height: '20px',
    },
    spacing: {
      paddingTop: theme.spacing(4),
    },
    noTxContainer: {
      padding: theme.spacing(4),
      display: 'flex',
      justifyContent: 'center',
    },
  })
);
