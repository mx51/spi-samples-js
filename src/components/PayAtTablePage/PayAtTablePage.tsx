import React from 'react';
import { Container, Typography, Button, Grid } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import Layout from '../Layout';
import { usePayAtTableStyles } from './PayAtTablePage.style';
import { Table } from './Table/Table';
import { useAppSelector } from '../../redux/hooks';
import { getAllTables } from '../../redux/reducers/PayAtTableSlice/payAtTableSelector';
import {
  addCostToATable,
  addNewTable,
  closeTable,
  toggleLocked,
  updateOperatorId,
} from '../../redux/reducers/PayAtTableSlice/payAtTableSlice';
import spiService from '../../services/spiService';

const PayAtTable: React.FC = () => {
  const classes = usePayAtTableStyles();
  const tables = useAppSelector(getAllTables);
  const dispatch = useDispatch();

  return (
    <Layout>
      <Container className={classes.container} maxWidth="md">
        <Typography variant="h6" component="h1">
          Pay At Tables
        </Typography>
        <Grid container spacing={2}>
          {tables.map((table) => (
            <Grid key={table.tableId} item xs={4}>
              <Table
                onToggleLocked={() => {
                  dispatch(toggleLocked(table.tableId));
                }}
                onNewCostAdded={(cost) => {
                  dispatch(addCostToATable({ tableId: table.tableId, cost }));
                }}
                onClose={() => dispatch(closeTable(table.tableId))}
                locked={table.locked}
                title={table.label}
                totalAmount={table.totalAmount}
                dueAmount={table.outStandingAmount}
                operatorIds={
                  spiService.state.patConfig.operatorIdEnabled ? spiService.state.patConfig.allowedOperatorIds : []
                }
                selectedOperatorId={table.operatorId}
                onOperatorIdChanged={(operatorId) => {
                  dispatch(updateOperatorId({ tableId: table.tableId, operatorId }));
                }}
                payments={table.bill.payments}
              />
            </Grid>
          ))}
        </Grid>
      </Container>
      <Button
        onClick={() => {
          dispatch(addNewTable());
        }}
        className={classes.addButton}
        color="primary"
        variant="contained"
      >
        Add
      </Button>
    </Layout>
  );
};

export default PayAtTable;
