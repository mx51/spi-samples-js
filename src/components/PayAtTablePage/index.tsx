import PayAtTablePage from './PayAtTablePage';

export default PayAtTablePage;
