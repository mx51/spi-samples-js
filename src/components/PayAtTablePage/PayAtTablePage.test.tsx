import React from 'react';
import { cleanup } from '@testing-library/react';
import PayAtTablePage from '.';
import mockWithRedux from '../../utils/tests/common';

describe('Test <PayAtTablePage />', () => {
  afterEach(cleanup);

  test('should match PayAtTablePage snapshot test', () => {
    // Arrange
    const container = mockWithRedux(<PayAtTablePage />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
