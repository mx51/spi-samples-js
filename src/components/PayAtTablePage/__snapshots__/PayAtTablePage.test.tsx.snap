// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PayAtTablePage /> should match PayAtTablePage snapshot test 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-10"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-11 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <div
            class="MuiBox-root MuiBox-root-16 makeStyles-navbar-13"
          >
            <div
              class="MuiBox-root MuiBox-root-17"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-14 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                data-test-id="navbarMenuIcon"
                name="menuIcon"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <svg
                height="15"
                width="48"
              >
                PrimaryLogoIcon.svg
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthMd"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph"
      >
        Pay At Table
      </h1>
      <div
        class="MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-18 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <svg>
              UnpluggedIcon.svg
            </svg>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
            >
              You don't have any terminals paired yet.
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Fortunately, it's very easy to pair a terminal and start using Espresso POS.
            </p>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              href="/terminals"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Go to terminals
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
